### DISCUSSION TITLE
Whether and when libunwind support build with cmake and clang

### DISCUSSION THREAD

**Discussion by JackyWoo** [libunwind/libunwind#630] [Ideas]
Whether and when libunwind support build with cmake and clang

**Comment by bregma** [comment]
The [CI builds regularly use clang](https://github.com/libunwind/libunwind/actions/runs/4599622445/jobs/8125217281).

Since the autotools work well, I see no need to rewrite everything in cmake. Is there any real measurable advantage that outweight the cost of a rewrite?

**Comment by JackyWoo** [comment]
@bregma Thanks for your reply. 

In my case I wanna use libunwind with git submodule and my project uses cmake.   CMake is becoming more and more popular nowadays. So  it is wonderful if libunwind supports building with cmake.

**Comment by bertwesarg** [comment]
I'm using CMake-based projects as sub-projects in Autotools-based projects regularly. If this is not possible the other way around, then CMake is probably not a a good build tool to begin with?

**Reply by jeremy-rifkin** [reply]
It is possible to include an autotools-based project in cmake however it's nowhere near as nice as just being able to use the library through cmake natively. Additionally there are portability issues. Cmake seems to be the widely accepted standard for building libraries these days, with a [lot of benefits over other alternatives](https://stackoverflow.com/a/18291580/15675011).

**Comment by JackyWoo** [comment]
@bertwesarg thanks for your reply.  

I notice that now libunwind support Cmake in `Visual Studio`, so I guess maybe clang and gcc will be supported and may I know when.

https://github.com/libunwind/libunwind/blob/5010beb099cbfe508766eb5cb76ed9ad75a8d1f2/CMakeLists.txt#L96-L98

**Comment by jeremy-rifkin** [comment]
It would be really cool if cmake support could be added for more than just VS on windows. I am working on a stack trace library and it would be nice to be able to grab this library as a dependency with `FindPackage`.

I would be happy to do the work to add cmake support however I'd need help understanding what checks / configurations are needed to get the library built.

**Reply by bregma** [reply]
Feel free to propose a pull request, as long as you're willing to maintain the CMake system going forward and make sure it keeps up with the autotools configuration. Make sure you include CI builds for all the supported target architectures, at least on Linux.
